boot/cmdline.go

```diff
@@ -242,3 +243,50 @@ func observeSuccessfulCommandLineCompatBoot(model *asserts.Model, m *Modeenv) (*
 	newM.CurrentKernelCommandLines = bootCommandLines{cmdlineExpected}
 	return newM, nil
 }
+
+// observeCommandLineUpdate observes a pending kernel command line change caused
+// by an update of boot config. When needed, the modeenv is updated with a
+// candidate command line and the encryption keys are resealed. This helper
+// should be called right before updating the managed boot config.
+func observeCommandLineUpdate(model *asserts.Model) error {
+	// TODO:UC20: consider updating a recovery system command line
+
+	m, err := loadModeenv()
+	if err != nil {
+		return err
+	}
+
+	cmdlineBootedWith, err := osutil.KernelCommandLine()
+	if err != nil {
+		return err
+	}
+	// this is the current expected command line
```

Comment:
shouldn't we use what is in modeenv already and append to it?  instead

Reply:
Updated